test/integration/test_plugin.cpp

```diff
@@ -9,25 +9,32 @@
 
 #include "gtest/gtest.h"
 
-int plugin_test_counter_pre{0};
-int plugin_test_counter_post{0};
+int plugin_test_capture_counter_pre{0};
+int plugin_test_capture_counter_post{0};
 
-// Check that the plugin is called the correct number of times, once before and
-// after each kernel invocation
+int plugin_test_launch_counter_pre{0};
+int plugin_test_launch_counter_post{0};
+
+// Check that the plugin is called the correct number of times,
+// once before and after each kernel capture for the capture counter
+// once before and after each kernel invocation for the launch counter
 TEST(PluginTest, Counter)
 {
   int* a = new int[10];
 
   for (int i = 0; i < 10; i++) {
     RAJA::forall<RAJA::seq_exec>(
-      RAJA::RangeSegment(0,10), 
+      RAJA::RangeSegment(0,10),
       [=] (int i) {
         a[i] = 0;
     });
```

Comment:
We should probably expand testing to different forall types to ensure the plugins are called the expected number of times.

Reply:
When I expanded testing I discovered that the forall(pointer, len, body) version was inaccessible.
Then discovered that camp::type_traits::is_iterator<int> appears to be true so the overload was disabled anyway.
I think we managed to deprecate this version of forall pretty effectively, should we remove it now since people probably couldn't have used it anyway? The forall_Icount(pointer, len, icount, body) version has the same problem with type_traits.

Reply:
@MrBurmark We officially deprecated the forall methods that take a length in favor of requiring all to take a segment a couple of releases ago. Did we not excise all those methods?

Reply:
They still existed in pattern/forall, but I removed them in this PR to simplify things a bit. Do you know if they are still references to them lurking in the docs somewhere?

Reply:
I think I removed them all. I will double-check.